src/respend/respendrelayer.cpp

```diff
@@ -86,7 +86,16 @@ void RespendRelayer::Trigger()
     if (!valid || !interesting)
         return;
 
-    RelayTransaction(respend);
+    CInv inv(MSG_TX, respend.GetHash());
+    LOCK(cs_vNodes);
+    for (CNode *pnode : vNodes)
+    {
+        if (!pnode->fRelayTxes || !pnode->fTestsRespends)
+            continue;
+        LOCK(pnode->cs_filter);
+        if (!pnode->pfilter || pnode->pfilter->IsRelevantAndUpdate(respend))
+            pnode->PushInventory(inv);
+    }
```

Comment:
this is changing the semantics cause `RelayTransaction` check also for max txn len:

```c++
    uint64_t len = ::GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION);
    if (len > maxTxSize.value)
    {
        LOGA("Will not announce (INV) excessive transaction %s.  Size: %llu, Limit: %llu\n", tx.GetHash().ToString(),
            len, (uint64_t)maxTxSize.value);
        return;
    }
```

Reply:
The alternative is to change RelayTransaction (or copy much more code), as such I went for the easiest solution. I'm open to suggestions (or just change and push, if you want).

I gave it a bit of thought and considered that the not forwarding of a double spend based on size is going to be very easy to abuse, better not filter on that.

If you disagree, let me know (or just fix it and push the way you prefer it instead).